### PR TITLE
feat: Move from Cookiecutter to Cookieninja

### PR DESCRIPTION
**Pull request by YurelyCamacho**
## Pull Request description
Change from Cookiecutter to Cookieninja package. Update doc, dependencies, tests.

Solve #33 

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [ x ] new feature
- [ ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ ] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [ ] New and old tests passed locally.

## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -48,14 +48,16 @@ isort = "^5"
...
-cookiecutter = "^1.7.3"
 pytest-cov = "^3.0.0"
 pytest-cookies = "^0.6.1"
 Sphinx = "^4.4.0"
 Flake8-pyproject = "^0.9.0"
 jupyter-book = "^0.12.3"
 myst-parser = "^0.15"
 
+[tool.poetry.group.dev.dependencies]
+cookieninja = "1.0.0"
```
Comment (xmnlab) [inline pyproject.toml:59]: thinking a bit about that .. cookieninja should not be a dev dep .. it should be a direct dependency

**Comment by xmnlab** [comment]
not sure why the CI failed .. but first move the cookieninja from dev to the direct dependency

do it using poetry command .. probably you need first to remove the dep and add it again

File: pyproject.toml (hunk 1 above)
Comment (YurelyCamacho) [reply pyproject.toml:59]: Ok, I'm doing it now

**Comment by YurelyCamacho** [comment]
Done

**Comment by xmnlab** [comment]
could you ensure that you are working on top the latest changes from upstream please?

maybe something like:

```
git fetch --all
git rebase -i upstream/main
```

# NOTE: JUST IF THERE ARE CONFLICTS

if there is any conflict when you are rebasing it, fix any conflict and run:

```
git add -u
git rebase --continue
```

PS: not sure if you are using a git prompt integration ...  if not .. check this one: https://github.com/magicmonty/bash-git-prompt
maybe it would help you to have more information about the repo status

**Comment by YurelyCamacho** [comment]
@xmnlab I have done what you asked me to do.

**Comment by xmnlab** [comment]
there is not problem with the latest version on upstream/main
so I guess it is a problem with the cookiecutter
you will need to debug it locally in order to find where this problem is located.

**Review by xmnlab — APPROVED** [review]
LGTM! thanks for working on that!

**Pull request merged**